internal/contentmanager/cache.go

```diff
@@ -0,0 +1,115 @@
+package contentmanager
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Cache is a storage mechanism for keeping track of
+// managed content sources. It also exposes methods
+// for retrieving the stored references of the managed
+// content
+type Cache interface {
+	// Get attempts to fetch the object with the provided namespace/name and GroupVersionKind
+	Get(client.ObjectKey, schema.GroupVersionKind) (runtime.Object, error)
+	// List attempts to list all the objects with the provided GroupVersionKind and label selector
+	List(schema.GroupVersionKind, labels.Selector) ([]runtime.Object, error)
```

Comment:
Why not use `client.Reader` here?

Reply:
Since nothing currently calls Get/List lets drop these functions and re-implement them when we have a concrete use case and better understanding of the calls a user may want to make.